content/open-source/wolfi/building-a-wolfi-package.md

```diff
@@ -0,0 +1,437 @@
+
+If you aren't very familiar with building packages from source using tools such as `cmake` and `autoconf`, it's a good idea to compile the package locally first - you don't need to run "make install"  at the end to get the package installed on your own system, but running the configure and make processes will give you a better understanding of the build requirements and configure options.
+## The melange YAML File
+The melange YAML file is where you'll define the details about the package and its build pipeline. If you are familiar with GitHub Actions, you'll find out that melange definitions are very similar to GitHub Actions workflows.
+### The `package` Section
+
+The melange YAML file starts with a `package` section, used to define metadata information and runtime dependencies. The following excerpt demonstrates how this section is declared in the Wolfi [PHP package](https://github.com/wolfi-dev/os/blob/34aa71ac4898c2b7c529548eafa51b0ea7a4dbd3/php.yaml#L1C3-L1C3) YAML:
+
+```yaml
+package:
+  name: php
+  version: 8.2.8
+  epoch: 0
+  description: the PHP programming language
+  copyright:
+    - license: PHP-3.01
+  dependencies:
+    runtime:
+      - libxml2
+```
+
+- **Package name**: convention is to use the same name as the YAML file without extension. This is what people will search for, so it's a good idea to keep it consistent with how the package is named in other distributions.
+- **Description**: this information shows up when searching for the package with apk.
+- **Version**: the version of the package.
+- **Epoch**: this is a numeric field set to zero by default, that only needs to be incremented when there is a non-version change in the package. For instance, when build options such as compiler flags have changed or new subpackages have been added but the upstream package version hasn't changed - in such cases, you'd need to "bump the epoch" in order to trigger the build.
+- **License**: the package license. It is important to note that only packages with OSI-approved licenses can be included in Wolfi. You can check the relevant package info in the [licenses page at opensource.org](https://opensource.org/licenses/).
+- **Runtime dependencies**: any dependencies needed by your package at runtime. Not to be confused with build dependencies, these will come up in the environment section of the file.
+
+### The `environment` Section
+
+The next section is the environment section. It defines how the build environment should look in order to build your package. Packages listed in this section won't be included in the final package, because they are only needed at build time.
```

Comment:
```suggestion
The next section is the `environment` section. It defines how the build environment should look in order to build your package. Packages listed in this section won't be included in the final package, because they are only needed at build time.
```

just for consistency